### PR TITLE
Get ABI from Sourcify

### PR DESCRIPTION
**Pull request by hasparus**
### What's changed?

I added two ways to grab metadata from Sourcify. It's not production-ready yet, but we can work with it.

### What needs fixing here?

`@ethereum-sourcify/contract-call-decoder` built code is ESM, so I had to make some changes to our test setup and change "type" to "module" in package.json files. This obviously cannot be published, because it would break everything. We can find a work around to use the decoder or maybe leverage `.mts` extension from TypeScript 4.5

### Questions

- What if contract is not verified on Sourcify?
- Using Abis vs Outputs (with NatSpec)?

### Further work

- Configuring sourcify vs etherscan in config?

### CODE REVIEW

**Comment by krzkaczor** [comment]
Nice work!

> What if contract is not verified on Sourcify?
I think we could implement it in this way:

1. Have config setting something like `abi-source: 'sourcify' | 'etherscan' | 'manual' (abi needs to be provided manually)`
2. If the contract is not verified on sourcify then simply throw an error. I would assume that users would prefer to use sourcify when they don't trust etherscan so I am not sure if automatic fallback to etherscan makes any sense.

>Using Abis vs Outputs (with NatSpec)?
We can ignore NatSpec entirely for now -- just use ABI.


I think we should prefer to use official SDK and if something's missing we should help to shape it. CC: @kuzdogan

**Comment by hasparus** [comment]
Hey @kuzdogan, I talked with @krzkaczor now, and we'll use the `sourcify.dev` endpoint for now — it's exactly what we need for eth-sdk (and we already keep a record of chainIds). This means we won't need that CJS support.

**Comment by hasparus** [comment]
@krzkaczor how would you see `abiSource: 'manual'`? Instead of fetching ABIs we could print a warning explaining how to get them?

**Comment by krzkaczor** [comment]
@hasparus 
> @krzkaczor how would you see abiSource: 'manual'? Instead of fetching ABIs we could print a warning explaining how to get them?

More or less yes. Simply put if ABI doesn't exist already just fail with instructions on how to provide it.

**Comment by hasparus** [comment]
Since we allow arbitrary network ids, I'm fixing names we currently don't handle.

image

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,5 @@ dist
 yarn-error.log
 .DS_Store
 *.tsbuildinfo
-packages/tests/eth-sdk
```
Comment (hasparus) [inline .gitignore:6]: Moved to a separate gitignore file

**Comment by hasparus** [comment]
All right, this is ready for review.

File: packages/eth-sdk/src/config/readConfig.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ async function registerTsNode() {
   try {
     const { register } = await import('ts-node')
 
-    register({ compilerOptions: { module: 'CommonJS' } })
+    register({ compilerOptions: { module: 'CommonJS' }, transpileOnly: true })
```
Comment (krzkaczor) [inline packages/eth-sdk/src/config/readConfig.ts:34]: I think we should warn users about problems with the config. Is there any reason to have this turned on? Did you figure out the problem with the "double" transpilation?

**Review by krzkaczor — COMMENTED** [review]
Missing changeset

**Review by krzkaczor — APPROVED** [review]

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**